aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -167,6 +167,17 @@ static void prod_kernel_impl(TensorIterator& iter) {
   });
 }
 
+// TODO: Implement nanprod similar to prod kernel
+static void nanprod_kernel_impl(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "nanprod_cpu", [&] {
```

Comment:
Note: Complex support is not added in this PR for CPU/CUDA devices. Only floating types for CPU device.

Reply:
What types does prod support, and what's the challenge with supporting all those types? For both CPU and CUDA doesn't the device-generic code in ReduceOps handle the non-float and non-complex case by just calling into prod?

By the way, a trick a recent PR used for autograd was to register a function, "foo", that called "bar" for integer inputs and "cat" for float and complex inputs. This let autograd pick up the existing function bar's autograd for integer inputs and cat's autograd, otherwise. In this case, for example, it would let you use prod's autograd for the non-float and non-complex cases, then you could have a function _nanprod with autograd registered that would handle the float and complex cases. Just an idea, I'm not sure how relevant it will be to this PR.